### ISSUE TITLE
Two types of Word2vec model 

### ISSUE THREAD

**Issue by WatershedXiaolan** [WatershedXiaolan/Xiaolan-s-Gitbook#356]
CBOW: Continuous bag of words (CBOW) tries to predict the current word from its surrounding words by optimizing for following loss function:
<img width="559" alt="Screen Shot 2022-03-31 at 8 29 15 PM" src="https://user-images.githubusercontent.com/5552371/161177108-ae52646c-05bf-47ec-8d6b-e9f3be530060.png">
<img width="813" alt="Screen Shot 2022-03-31 at 8 29 47 PM" src="https://user-images.githubusercontent.com/5552371/161177163-0a590826-6a5a-4629-803c-3d0364c3e5f4.png">

**Comment by WatershedXiaolan** [comment]
Skipgram: In this architecture, we try to predict surrounding words from the current word. The loss function will now look like:
<img width="534" alt="Screen Shot 2022-03-31 at 8 30 17 PM" src="https://user-images.githubusercontent.com/5552371/161177198-89373fc9-46b7-4934-a838-eceb7b1826ef.png">
<img width="791" alt="Screen Shot 2022-03-31 at 8 30 30 PM" src="https://user-images.githubusercontent.com/5552371/161177233-4c318741-4ba9-45c5-9756-87bce0cd83e2.png">

**Comment by WatershedXiaolan** [comment]
CBOW is used for smaller training set 
Skipgram is used for larger training set
